_faq/cep-and-its-importance.md

```diff
@@ -0,0 +1,10 @@
+---
+question: "What is a cEP and why it is so important to do analysis pre-gsoc and design as early as possible, to be finalized in the cEP?"
+---
+A **cEP(coala Enhancement Proposal)** is a proposal proposed to the coala community discussing the new major features that the contributor is intended to implement during GSoC.
+
+**cEP's** are designed before implementation. Implementing major features usually happens from a technical perspective and may lead to not well-designed content - developer and user wise. This can essentially save a lot of coding work when big features are written which are oftentimes not well enough designed initially. This also makes sure that major features are properly documented.
+
+Anyone planning to do GSoC with coala should be starting to write detailed project ideas in our [projects](https://github.com/coala/projects) repository. Experience tells us that the best GSoC students are the ones who heavily contributed to the project ideas website, creating projects from the issues, and expanding on existing ***.md*** files. And generally learning how to express complex technical ideas in the markup. If you can't explain your idea, and how you will achieve it, we probably don't want to mentor you doing that undefined project.
```

Comment:
`And generally learning how to express complex...` is a sentence fragment, could either reword so that it stands alone as a complete sentence or attach it to the previous sentence.